.github/scripts/gha-functions.sh

```diff
@@ -0,0 +1,33 @@
+#!/bin/bash
+
+# ------------------------------------------------------------------------------
+#
+# Description:
+#   This file contains reusable utility functions for GitHub Actions workflows.
+#
+#   You can source this file in any workflow step to make the functions
+#   available within that step.
+#
+# Usage:
+#   source .github/scripts/gha-functions.sh
+#   export_new_env_vars path/to/your/script.sh
+# ------------------------------------------------------------------------------
+
+export_new_env_vars() {
```

Comment:
could we add a description of what the purpose of this function is?

It's a bit hard to read, and I can imagine that looking back at this in a couple of weeks will require new deciphering.

In the end it adds variables that are currently not part of the env of the step as env for subsequent steps by adding those to the `GITHUB_ENV` as you probably rely on the new env vars somewhere and don't want to re-export them / having to think about it.

I think it might be slightly deceptive since env != GITHUB_ENV but that's probably not the purpose of this function.